### PR TITLE
implemented check for namespace and servicetemplate id, refs #278

### PR DESCRIPTION
**Pull request by przytads**
any comments?

### CODE REVIEW

**Comment by eiselems** [comment]
I like the implementation especially the HashMap approach.

But i think it has a major flaw, what does the Query really do?
Does it select the <ServiceTemplate>'s targetnamespace and id attribute?

**Comment by przytads** [comment]
Whats the "major flaw"? It selects the attributes id and targetNamespace from the element ServiceTemplate.

**Comment by przytads** [comment]
I removed the namespace check.

File: src/main/java/org/opentosca/csarrepo/service/UploadCsarFileService.java [hunk 1]
```diff
@@ -85,6 +101,33 @@ private void storeFile(long csarId, InputStream inputStream, String name) {
...
+			}
+			String serviceTemplateId = nodeMap.get("id");
+			String namespace = nodeMap.get("targetNamespace");
+
+			if (null == csar.getServiceTemplateId()) {
+				csar.setServiceTemplateId(serviceTemplateId);
+				csar.setNamespace(namespace);
+				LOGGER.info("csar: service template id ({}) and namespace ({}) set", serviceTemplateId, namespace);
+			} else if (!csar.getServiceTemplateId().equals(serviceTemplateId)
+					|| (null == csar.getNamespace() && null != namespace && !namespace.equals(null))
```
Comment (eiselems) [inline src/main/java/org/opentosca/csarrepo/service/UploadCsarFileService.java]: I don't want to get fussy here but I think the !namespace.equals(null) is obsolete here

Comment (przytads) [inline src/main/java/org/opentosca/csarrepo/service/UploadCsarFileService.java]: true :+1:

**Pull request merged**